services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -1928,6 +1935,101 @@ export class WorkflowService {
         },
       });
 
+      service.subscribe(WorkflowEvents.ENTITIES_UPDATE, async ({ payload }) => {
```

Comment:
The code here must be in a different location. It makes the `createOrUpdateWorkflowRuntime` much less readable than it already is.